client/internal/storage/repository/syncevent.go

```diff
@@ -0,0 +1,37 @@
+package repository
+
+import (
+	"client/internal/logger"
+	"context"
+	"database/sql"
+)
+
+// TODO: в перспективе добавить еще и пользователя в эвент.
+const getLastSyncEventQuery = `
+  SELECT version from sync_events where kind = $1 order by version desc limit 1
+`
+const saveSyncEvent = `INSERT INTO sync_events (kind, version) VALUES ($1, $2)`
+
+func (si *Storage) GetLastSyncEventVersion(ctx context.Context, kind string) (version string, err error) {
+	row := si.driver.QueryRow(getLastSyncEventQuery, kind)
+	err = row.Scan(&version)
+
+	if err == nil {
+		return version, nil
+	}
+
+	if err == sql.ErrNoRows {
+		return "0", nil
+	} else {
+		return "", err
+	}
+}
```

Comment:
Стоит организовывать код по приципу `early return`, коггда мы проверяем именно наличе ошибки, не не ее отсутствие.
```suggestion
func (si *Storage) GetLastSyncEventVersion(ctx context.Context, kind string) (version string, err error) {
	row := si.driver.QueryRow(getLastSyncEventQuery, kind)
	err = row.Scan(&version)

	if err != nil {
		if err == sql.ErrNoRows {
			return "0", nil
		}
		return "", err
	}

	return version, nil
}
```

Reply:
Согласен. Поправил в двух местах